core/meta/src/TClass.cxx

```diff
@@ -3451,38 +3457,31 @@ const char *TClass::GetSharedLibs()
 
 TList *TClass::GetListOfBases()
 {
-   if (!fBase) {
-      if (fCanLoadClassInfo) {
-         if (fState == kHasTClassInit) {
+   if (fBase)
+      return fBase;
 
-            R__LOCKGUARD(gInterpreterMutex);
-            // NOTE: Add test to prevent redo if another thread has already done the work.
-            // if (!fHasRootPcmInfo) {
+   if (fState == kHasTClassInit) {
+      R__LOCKGUARD(gInterpreterMutex);
 
-            // The bases are in our ProtoClass; we don't need the class info.
-            TProtoClass *proto = TClassTable::GetProtoNorm(GetName());
-            if (proto && proto->FillTClass(this)) {
-               // Not sure this code is still needed
-               // R__ASSERT(kFALSE);
+      // The bases are in our ProtoClass; we don't need the class info.
+      TProtoClass *proto = TClassTable::GetProtoNorm(GetName());
+      if (proto && proto->FillTClass(this))
+         fHasRootPcmInfo = true;
+   }
 
-               fHasRootPcmInfo = kTRUE;
-            }
-         }
-         // We test again on fCanLoadClassInfo has another thread may have executed it.
-         if (!fHasRootPcmInfo && !fCanLoadClassInfo) {
-            LoadClassInfo();
```

Comment:
> This check made no sense to me, also because it mentions other threads doing the work, 

The existing code is
```
if (fCanLoadClassInfo) {
   if (...) {
      R__LOCKGUARD(...);
   }
   if (!fHasRootPcmInfo && !fCanLoadClassInfo) {
```
The comment says that in-between the first line and the last line, another thread may have successfully run LoadClassInfo and thus, even-though we already tested for the value of fCanLoadClassInfo, we should test for it again.


On another note the question is why did you remove the test on '!fHasRootPcmInfo'